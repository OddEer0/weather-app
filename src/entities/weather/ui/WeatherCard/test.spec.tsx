import { render, screen } from "@testing-library/react"

import { WeatherCard } from "./index"

describe("Should render weather card", () => {
	it("Should render component", () => {
		render(<WeatherCard date="Date text" temperature="14C" status={1000} />)
		expect(screen.getByText(/date text/i)).toBeInTheDocument()
		expect(screen.getByText(/14C/i)).toBeInTheDocument()
	})

	it("Should render component with after props", () => {
		render(<WeatherCard date="Date text" temperature="14C" status={1000} after="5C" />)
		expect(screen.getByText(/5C/i)).toBeInTheDocument()
	})
})

describe("Screenshot testing component", () => {
	it("Screenshot component", () => {
		const { asFragment } = render(<WeatherCard date="Date text" temperature="14C" status={1000} />)
		expect(asFragment()).toMatchSnapshot()
	})
})
