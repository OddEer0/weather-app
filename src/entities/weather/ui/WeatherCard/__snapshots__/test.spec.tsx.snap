// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Screenshot testing component > Screenshot component 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root _card_6a843d css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root _content_6a843d css-46bh2p-MuiCardContent-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
      >
        Date text
      </span>
      <div
        class="_icon_6a843d"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"
          />
        </svg>
      </div>
      <div
        class="_footer_6a843d _footerCenter_6a843d"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
        >
          14C
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
